### ISSUE TITLE
"IndexError: list index out of range" with long prompts and wrong syntax fractional numbers

### ISSUE THREAD

**Issue by willezgo** [damian0815/compel#11]
Hello, first off, thank you for this awesome lib, great work!

I stumbled upon this bug in the latest pip version (1.0.2).

It seems to happen with the following conditions:

- Long prompt
- Presence of a fractional number that use the wrong syntax, like Auto1111. Ex: (word:1.5)

Reproduction code:

```py
from transformers import *
from compel import Compel
import torch


tokenizer    = CLIPTokenizer.from_pretrained("runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16, subfolder="tokenizer")
text_encoder = CLIPTextModel.from_pretrained("runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16, subfolder="text_encoder").to("cuda")


neg_prompt = "ugly, tiling, poorly drawn hands, poorly drawn feet, poorly drawn face, out of frame, extra limbs, disfigured, deformed, body out of frame, blurry, bad anatomy, blurred, watermark, grainy, signature, cut off, draft, mutation, morbid, mutilated, mutilated hands, cloned face, gross proportions, missing arms, extra arms, missing legs, extra legs, extra feet, (ugly face:1.5)"

compel = Compel(tokenizer=tokenizer, text_encoder=text_encoder, truncate_long_prompts=True)
neg_embed = compel(neg_prompt)

print(neg_embed)
```

Stacktrace:

```
IndexError                                Traceback (most recent call last)
[<ipython-input-18-07bd5f2fa5e1>](https://localhost:8080/#) in <module>
     12 
     13 compel = Compel(tokenizer=tokenizer, text_encoder=text_encoder, truncate_long_prompts=True)
---> 14 neg_embed = compel(neg_prompt)
     15 
     16 print(neg_embed)

5 frames
[/usr/local/lib/python3.9/dist-packages/torch/autograd/grad_mode.py](https://localhost:8080/#) in decorate_context(*args, **kwargs)
     25         def decorate_context(*args, **kwargs):
     26             with self.clone():
---> 27                 return func(*args, **kwargs)
     28         return cast(F, decorate_context)
     29 

[/usr/local/lib/python3.9/dist-packages/compel/compel.py](https://localhost:8080/#) in __call__(self, text)
     78         cond_tensor = []
     79         for text_input in text:
---> 80             cond_tensor.append(self.build_conditioning_tensor(text_input))
     81 
     82         cond_tensor = torch.cat(cond_tensor)

[/usr/local/lib/python3.9/dist-packages/compel/compel.py](https://localhost:8080/#) in build_conditioning_tensor(self, text)
     68     def build_conditioning_tensor(self, text: str) -> torch.Tensor:
     69         prompt_object = self.parse_prompt_string(text)
---> 70         conditioning, _ = self.build_conditioning_tensor_for_prompt_object(prompt_object)
     71         return conditioning
     72 

[/usr/local/lib/python3.9/dist-packages/compel/compel.py](https://localhost:8080/#) in build_conditioning_tensor_for_prompt_object(self, prompt)
    115                 return cac_args.original_conditioning, { 'cross_attention_control': cac_args }
    116             else:
--> 117                 return self._get_conditioning_for_flattened_prompt(prompt), {}
    118 
    119         raise ValueError(f"unsupported prompt type: {type(prompt).__name__}")

[/usr/local/lib/python3.9/dist-packages/compel/compel.py](https://localhost:8080/#) in _get_conditioning_for_flattened_prompt(self, prompt, should_return_tokens)
    157         fragments = [x.text for x in prompt.children]
    158         weights = [x.weight for x in prompt.children]
--> 159         conditioning, tokens = self.conditioning_provider.get_embeddings_for_weighted_prompt_fragments(
    160             text_batch=[fragments], fragment_weights_batch=[weights], should_return_tokens=True, device=self.device)
    161         if should_return_tokens:

[/usr/local/lib/python3.9/dist-packages/compel/embeddings_provider.py](https://localhost:8080/#) in get_embeddings_for_weighted_prompt_fragments(self, text_batch, fragment_weights_batch, should_return_tokens, device)
    122                 if fragment_weight < 1:
    123                     if self.downweight_mode == DownweightMode.MASK:
--> 124                         fragment_start_token_id, fragment_end_token_id = fragment_token_index_ranges[index]
    125                         # mask out this fragment
    126                         mask_without_fragment = mask.clone()

IndexError: list index out of range
```

**Comment by damian0815** [comment]
omg are you the *Very Nervous System* David Rokeby?? that piece thoroughly and richly informed my view about using computer vision for art! 🙇 🙇 🙇 🙇 🙇 🙇 🙇 🙇 🙇 🙇 🙇 🙇 🙇

**Comment by damian0815** [comment]
@willezgo @david-rokeby bug should be fixed with compel 1.0.5 - sorry it took me a few weeks to get to it.
